include/vapor/Field.h

```diff
@@ -32,30 +33,33 @@ class FLOW_API Field
 
     //
     // Get the field value at a certain position, at a certain time.
-    // Users could control if this method checks position inside volume.
     //  
     virtual int  GetScalar(  float time, const glm::vec3& pos,  // input 
-                             float& val,                        // output
-                             bool checkInsideVolume = true ) const = 0;
+                             float& val ) const = 0;            // output
 
     //
     // Get the velocity value at a certain position, at a certain time.
-    // Users could control if this method checks position inside volume.
     //  
     virtual int  GetVelocity( float time, const glm::vec3& pos, // input 
-                              glm::vec3& vel,                   // output
-                              bool checkInsideVolume = true ) const = 0;
+                              glm::vec3& vel ) const = 0;       // output
 
     //
     // Returns the number of empty velocity variable names.
     // It is 3 when the object is newly created, or is used to represent a scalar field
     // 
     int GetNumOfEmptyVelocityNames() const;
 
+    //
+    // Provide an option to cache and lock certain parameters.
+    // Both functions return 0 on success.
+    //
+    virtual auto LockParams()   -> int = 0;
+    virtual auto UnlockParams() -> int = 0;
```

Comment:
Is there a reason this is written this way? It is inconsistent with the reset of the code base. Also is it confirmed that this works with the windows compiler?

Reply:
It's just a new syntax that facilitates some meta programming use cases. Here it simply looks different.

It's part of the standard, and a lot of people advocate for it (I kind of like it too). I'd be surprised if visual studio doesn't support. If that's the case, I'd have no problem to convert to the old style.

Reply:
C++11. This is something required to support multiple C++11 meta programming features, so it's unlikely to be left out.